lib/sqlalchemy/sql/compiler.py

```diff
@@ -1957,6 +1957,21 @@ def visit_tablesample(self, tablesample, asfrom=False, **kw):
 
         return text
 
+    def visit_values(self, element, asfrom=False, **kw):
+        v = "VALUES %s" % ", ".join(
+            self.process(elem, literal_binds=True) for elem in element.list
+        )
+        if asfrom:
+            if element.alias_name:
+                v = "(%s) AS %s (%s)" % (
+                    v,
+                    element.alias_name,
+                    (", ".join(c.name for c in element.columns)),
```

Comment:
I'm just working using the recipe at https://github.com/sqlalchemy/sqlalchemy/wiki/PGValues and discovered this PR. Should these column names be compiled, in case they have spaces in the names? This is a problem I have just encountered with the recipe.

Reply:
yes good point.  with the columns, when we have a name already, we typically use the identifier preparer to set this up but I guess we could process the whole Column object too.

Reply:
needs a test case with some column names that require quoting.

Reply:
I've been trying to use the recipe for a table melt on Postgresql and find that when I get a values clause and apply a `lateral` alias, the produced syntax is not quite right. I thought worth mentioning in case it might highlight an issue here. The syntax produced was e.g. `FROM LATERAL (VALUES (1, 'textA', 99), (2, 'textB', 88)) AS t(mykey, mytext, myint) AS t_1` (or without the values alias). The actual syntax required would be as above without the final t_1 alias which has been applied by the lateral()

Reply:
lateral is another weird one to support, yes.    can you illustrate the SQL staement you want to see rendered?  I can work up a test case from that.

Reply:
Here is how I was trying to use the values() and lateral(), with the output and desired output at the bottom. https://gist.github.com/rad-pat/5c926e40600724597841a7ec211ba25f    My solution/hack for the moment is to create the values() with an is_lateral kwarg, set the _is_lateral of the FromClause and then write out the LATERAL is the kwarg is set.

Reply:
I've addressed both concerns in the latest patch at 
https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/1608

Reply:
Yep, like it'll do the job to me. Thanks